src/plugins/dashboard/public/application/components/dashboard_listing.tsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import { i18n } from '@osd/i18n';
+import { useMount } from 'react-use';
+import {
+  useOpenSearchDashboards,
+  TableListView,
+} from '../../../../opensearch_dashboards_react/public';
+import { CreateButton } from '../listing/create_button';
+import { DashboardConstants } from '../../dashboard_constants';
+import { DashboardServices } from '../../types';
+import { getTableColumns } from '../utils/get_table_columns';
+import { getNoItemsMessage } from '../utils/get_no_items_message';
+
+export const EMPTY_FILTER = '';
+
+export const DashboardListing = () => {
```

Comment:
This is already an improvement from the angular stuff.

Do we want to break this component down even further? I bring this up because there is a lot of specific functions in this component that is in the DashboardListing const.

I think a bug like this: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4017

Could have been easily caught if there were some unit tests behind the functions within this component. Should we think about creating an issue to break down this large components into units so that we can write unit tests on them to verify inputs and outputs?

Reply:
Sure i can create an issue for this task, and added to the backlog.

Reply:
#4051